### ISSUE TITLE
Debug_traceBlockByHash shows output for tx hash as "Execution Reverted" falsely

### ISSUE THREAD

**Issue by jwelch-qn** [ethereum/go-ethereum#29778]
#### System information

Geth version: `1.14.2`
CL client & version: `lighthouse, but N/A`
OS & Version: `Ubuntu 22`
Commit hash : N/A

Example: 
blockHash: `0xa8f23b3c94c8c5156dac46b93f1575f2c3ff7bc35d07276056fb5745d550875f`
txHash: `0x54ee3181242a78fb55c484f6de1c58aacb2e7abc08c8a4e25f799ae47dcd964d`

#### Expected behaviour

This tx is successful according to etherscan and other clients (Erigon, Nethermind, and older Geth)
https://etherscan.io/tx/0x54ee3181242a78fb55c484f6de1c58aacb2e7abc08c8a4e25f799ae47dcd964d

Geth 1.13.15
```
    {
      "txHash": "0x54ee3181242a78fb55c484f6de1c58aacb2e7abc08c8a4e25f799ae47dcd964d",
      "result": {
        "from": "0xc252ceec3ad24c47657638c323d96249a9f1b42a",
        "gas": "0xa1284",
        "gasUsed": "0x28061",
        "to": "0xd42b0ecf8a9f8ba9db7b0c989d73cf0bd5f83b66",
        "input": "0x05f497df75e26b9977ebb3a4d68a155e846020979f11b30ad133b381c78f24c469c46dfc970000000000000000000000007635831f45dc9c9d0d0b81d29ef725fb5dcabf4e000000000000000000000000000000000000000000000000000000000000000000000
0000000000000000000a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2000000000000000000000000000000000000000000000000000000000204583d0000000000000000000000000000000000000000
0000000000206e9f2b26bfb08a8000000000000000000000b8f81bb640006643c29500000000000000000000b69c88fc9fca744e7f0aaba70e988204f9bdc9a15ba061ece073affdc2b67bb3c6bb074ae9430585ff71d4c78d8e07d39d5b6289a24d8253194032e78758c3ca00000000000
0000000000000000000000000000000000000000000000204583da80000e100004000000000000000000000000000000000000022634d94f7fa8200000000000000000000000000000000000000000000000000000000000001a00000000000000000000000000000000000000000000000
000000000000000135a263f4e84902997eff4537d4d0aaefbb4ca5ac95000000c1000000540000005400000054000000540000002a0000000000000000fb2809a5314473e1165f6b58018e20ed8f07b8402024b700001e806643c1d50000b42ae51d2024b700b4fb2809a5314473e1165f6
b58018e20ed8f07b8402024b700001e806643c1d50000b42ae51d2024b700b4fb2809a5314473e1165f6b58018e20ed8f07b8406643c1bdb09498030ae3416b66dc000080af6840355608efd9a900000c989d73cf0bd5f83b660000d18bd45f0b94f54a968f0000d61b892b2ad6249011850000a59ca88d5813e693528f0000299aad897099c9fc85ce000038d42b0ecf8a9f8ba9db7b0c989d73cf0bd5f83b66010101a263f4e84902997eff4537d4d0aaefbb4ca5ac9500130500002100000000000000002935426dcb0c0e0000000000000000000000",
        "calls": [
          {
            "from": "0xd42b0ecf8a9f8ba9db7b0c989d73cf0bd5f83b66",
            "gas": "0x965d4",
            "gasUsed": "0x25a34",
            "to": "0x5b93d80da1a359340d1f339fb574bdc56763f995",
            "input": "0x05f497df75e26b9977ebb3a4d68a155e846020979f11b30ad133b381c78f24c469c46dfc970000000000000000000000007635831f45dc9c9d0d0b81d29ef725fb5dcabf4e0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2000000000000000000000000000000000000000000000000000000000204583d00000000000000000000000000000000000000000000000000206e9f2b26bfb08a8000000000000000000000b8f81bb640006643c29500000000000000000000b69c88fc9fca744e7f0aaba70e988204f9bdc9a15ba061ece073affdc2b67bb3c6bb074ae9430585ff71d4c78d8e07d39d5b6289a24d8253194032e78758c3ca000000000000000000000000000000000000000000000000000000000204583da80000e100004000000000000000000000000000000000000022634d94f7fa8200000000000000000000000000000000000000000000000000000000000001a0000000000000000000000000000000000000000000
0000000000000000000135a263f4e84902997eff4537d4d0aaefbb4ca5ac95000000c1000000540000005400000054000000540000002a0000000000000000fb2809a5314473e1165f6b58018e20ed8f07b8402024b700001e806643c1d50000b42ae51d2024b700b4fb2809a5314473e1165f6b58018e20ed8f07b8402024b700001e806643c1d50000b42ae51d2024b700b4fb2809a5314473e1165f6b58018e20ed8f07b8406643c1bdb09498030ae3416b66dc000080af6840355608efd9a900000c989d73cf0bd5f83b660000d18bd45f0b94f54a968f0000d61b892b2ad62490
11850000a59ca88d5813e693528f0000299aad897099c9fc85ce000038d42b0ecf8a9f8ba9db7b0c989d73cf0bd5f83b66010101a263f4e84902997eff4537d4d0aaefbb4ca5ac9500130500002100000000000000002935426dcb0c0e0000000000000000000000",
            "calls": [
              {
                "from": "0xd42b0ecf8a9f8ba9db7b0c989d73cf0bd5f83b66",
                "gas": "0x9338e",
                "gasUsed": "0x24d18",
                "to": "0x111111125421ca6dc452d289314280a0f8842a65",
                "input": "0xf497df75e26b9977ebb3a4d68a155e846020979f11b30ad133b381c78f24c469c46dfc970000000000000000000000007635831f45dc9c9d0d0b81d29ef725fb5dcabf4e000000000000000000000000000000000000000000000000000000000000000
//...
```

#### Actual behaviour
Geth, as of 1.14.2, shows in 
```
    {
      "txHash": "0x54ee3181242a78fb55c484f6de1c58aacb2e7abc08c8a4e25f799ae47dcd964d",
      "result": {
        "from": "0xc252ceec3ad24c47657638c323d96249a9f1b42a",
        "gas": "0xa1284",
        "gasUsed": "0xd44a",
        "to": "0xd42b0ecf8a9f8ba9db7b0c989d73cf0bd5f83b66",
        "input": "0x05f497df75e26b9977ebb3a4d68a155e846020979f11b30ad133b381c78f24c469c46dfc970000000000000000000000007635831f45dc9c9d0d0b81d29ef725fb5dcabf4e0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2000000000000000000000000000000000000000000000000000000000204583d00000000000000000000000000000000000000000000000000206e9f2b26bfb08a8000000000000000000000b8f81bb640006643c29500000000000000000000b69c88fc9fca744e7f0aaba70e988204f9bdc9a15ba061ece073affdc2b67bb3c6bb074ae9430585ff71d4c78d8e07d39d5b6289a24d8253194032e78758c3ca000000000000000000000000000000000000000000000000000000000204583da80000e100004000000000000000000000000000000000000022634d94f7fa8200000000000000000000000000000000000000000000000000000000000001a00000000000000000000000000000000000000000000000000000000000000135a263f4e84902997eff4537d4d0aaefbb4ca5ac95000000c1000000540000005400000054000000540000002a0000000000000000fb2809a5314473e1165f6b58018e20ed8f07b8402024b700001e806643c1d50000b42ae51d2024b700b4fb2809a5314473e1165f6b58018e20ed8f07b8402024b700001e806643c1d50000b42ae51d2024b700b4fb2809a5314473e1165f6b58018e20ed8f07b8406643c1bdb09498030ae3416b66dc000080af6840355608efd9a900000c989d73cf0bd5f83b660000d18bd45f0b94f54a968f0000d61b892b2ad6249011850000a59ca88d5813e693528f0000299aad897099c9fc85ce000038d42b0ecf8a9f8ba9db7b0c989d73cf0bd5f83b66010101a263f4e84902997eff4537d4d0aaefbb4ca5ac9500130500002100000000000000002935426dcb0c0e0000000000000000000000",
        "output": "0xfb8ae129",
        "error": "execution reverted",
        "calls": [
          {
            "from": "0xd42b0ecf8a9f8ba9db7b0c989d73cf0bd5f83b66",
            "gas": "0x965d4",
            "gasUsed": "0x4d9d",
            "to": "0x5b93d80da1a359340d1f339fb574bdc56763f995",
            "input": "0x05f497df75e26b9977ebb3a4d68a155e846020979f11b30ad133b381c78f24c469c46dfc970000000000000000000000007635831f45dc9c9d0d0b81d29ef725fb5dcabf4e0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000a0b86991c6218b36c1d19d4a2e9eb0ce3606eb48000000000000000000000000c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2000000000000000000000000000000000000000000000000000000000204583d00000000000000000000000000000000000000000000000000206e9f2b26bfb08a8000000000000000000000b8f81bb640006643c29500000000000000000000b69c88fc9fca744e7f0aaba70e988204f9bdc9a15ba061ece073affdc2b67bb3c6bb074ae9430585ff71d4c78d8e07d39d5b6289a24d8253194032e78758c3ca000000000000000000000000000000000000000000000000000000000204583da80000e100004000000000000000000000000000000000000022634d94f7fa8200000000000000000000000000000000000000000000000000000000000001a00000000000000000000000000000000000000000000000000000000000000135a263f4e84902997eff4537d4d0aaefbb4ca5ac95000000c1000000540000005400000054000000540000002a0000000000000000fb2809a5314473e1165f6b58018e20ed8f07b8402024b700001e806643c1d50000b42ae51d2024b700b4fb2809a5314473e1165f6b58018e20ed8f07b8402024b700001e806643c1d50000b42ae51d2024b700b4fb2809a5314473e1165f6b58018e20ed8f07b8406643c1bdb09498030ae3416b66dc000080af6840355608efd9a900000c989d73cf0bd5f83b660000d18bd45f0b94f54a968f0000d61b892b2ad6249011850000a59ca88d5813e693528f0000299aad897099c9fc85ce000038d42b0ecf8a9f8ba9db7b0c989d73cf0bd5f83b66010101a263f4e84902997eff4537d4d0aaefbb4ca5ac9500130500002100000000000000002935426dcb0c0e0000000000000000000000",
            "output": "0xfb8ae129",
            "error": "execution reverted",
            "calls": [
              {
                "from": "0xd42b0ecf8a9f8ba9db7b0c989d73cf0bd5f83b66",
                "gas": "0x9338e",
//...
```
#### Steps to reproduce the behaviour

Simply request this block hash in a `debug_traceBlockByHash` and check the txhash mentioned above.
```"jsonrpc": "2.0", "id": 1, "method": "debug_traceBlockByHash", "params": ["0xa8f23b3c94c8c5156dac46b93f1575f2c3ff7bc35d07276056fb5745d550875f",{"Tracer": "callTracer"}]}```

**Comment by quickchase** [comment]
I don't know if related to #29800 but we're also seeing errors on the following request:

```json
{
  "method": "debug_traceBlockByNumber",
  "params": [
    "0x12f7a28",
    {
      "tracer": "callTracer"
    }
  ],
  "id": 1,
  "jsonrpc": "2.0"
}
```

It's returning `insufficient funds` error

**Comment by s1na** [comment]
Top of my head this is the PR that changed traceBlock recently: https://github.com/ethereum/go-ethereum/pull/29402. Looking into it.

**Comment by s1na** [comment]
This should be fixed by #29811. Please feel free to re-open if issue persists.

**Comment by Mortiemi** [comment]
> This should be fixed by #29811. Please feel free to re-open if issue persists.
